app/policies/g5_updatable/location_policy.rb

```diff
@@ -0,0 +1,34 @@
+module G5Updatable
+  class LocationPolicy < G5Authenticatable::BasePolicy
+    class Scope < G5Authenticatable::BasePolicy::BaseScope
+
+      def resolve
+        return scope.all if has_global_role?
+        scope.where(id: location_roles.map(&:resource_id))
```

Comment:
Ok, Sounds good. I'll go with that.

On Thu, Feb 18, 2016 at 10:55 AM, Maeve Revels <notifications@github.com>
wrote:

> In app/policies/g5_updatable/location_policy.rb
> <https://github.com/G5/g5_authenticatable/pull/46#discussion_r53362887>:
>
> > @@ -0,0 +1,34 @@
> > +module G5Updatable
> > +  class LocationPolicy < G5Authenticatable::BasePolicy
> > +    class Scope < G5Authenticatable::BasePolicy::BaseScope
> > +
> > +      def resolve
> > +        return scope.all if has_global_role?
> > +        scope.where(id: location_roles.map(&:resource_id))
>
> In addition to considering user's location roles, I think this resolve
> method should also include locations associated with the user's client
> roles. If a user is assigned a role on Client A, then they should also have
> access to all of Client A's locations. This business rule holds true across
> all contexts (that I know of).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/G5/g5_authenticatable/pull/46/files#r53362887>.
>



-- 
Jonathan